canonical/vanilla-framework
# Add theme support for code snippet

## Done

- Added theme support for icons used in code snippets (get-link, linux-prompt)
- Added theme support for code snippets
- Added dark theme examples for code snippets: `/dropdown-multiple-dark`, `/default-dark`, `/icon-dark`

## Not done
- Code snippet with syntax highlighting is currently not supported on dark theme
- Cannot use the current theme colours because the dark code background is grey and not black, doesn't have the correct contrast in colour


Fixes [WD-8977](https://warthogs.atlassian.net/jira/software/c/projects/WD/boards/801?selectedIssue=WD-8977)

## QA

- Open [demo](https://vanilla-framework-5004.demos.haus/docs/examples/patterns/code-snippet/dropdown-multiple-dark)
- Go to `docs/examples` and check if code snippets' themes update accordingly
- Check examples:  
   - Code Snippet / Dark
   - Code Snippet / Multiple dropdowns Dark
   - Code Snippet / Icon Dark

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)


## Screenshots

<img width="1618" alt="image" src="https://github.com/canonical/vanilla-framework/assets/83575/74fb0622-2633-498e-b513-d7c939e4dbf7">



[WD-8977]: https://warthogs.atlassian.net/browse/WD-8977?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
